File: Core CS/Algorithms/Implementation/Between Two Sets/betweenTwoSets.kt
import java.io.*
import java.text.*
import java.util.*

fun gcd(a: Int, b: Int): Int{
    if (b == 0) 
        return a
    return gcd(b, a % b); 
}

fun gcd(a: Array<Int>): Int{
    var ans = a[0]
    for(i in 1 until a.size)
        ans = gcd(ans, a[i])
    return ans
}

fun lcm(a: Int, b: Int): Int{
    return a * (b / gcd(a, b))
}

fun lcm(a: Array<Int>): Int{
    var ans = a[0]
  
    for (i in 1 until a.size) 
        ans = (((a[i] * ans)) / (gcd(a[i], ans)))
  
    return ans; 
}

fun getTotalX(a: Array<Int>, b: Array<Int>): Int {
    val l = lcm(a)
    val g = gcd(b)
    
    var count = 0
    var i = l
    var j = 2
    while(i <= g){
        if(g%i==0)
            count++
            
        i = l * j
        j++
    }
    return count
}

fun main(args: Array<String>) {
    val first_multiple_input = readLine()!!.trimEnd().split(" ")

    val n = first_multiple_input[0].toInt()

    val m = first_multiple_input[1].toInt()

    val arr = readLine()!!.trimEnd().split(" ").map{ it.toInt() }.toTypedArray()

    val brr = readLine()!!.trimEnd().split(" ").map{ it.toInt() }.toTypedArray()

    val total = getTotalX(arr, brr)

    println(total)
}

File: Core CS/Algorithms/Implementation/Between Two Sets/betweenTwoSets.scala
import java.io._
import java.text._
import java.util._

object Result {
    
    def gcd(a: Int, b: Int): Int {
        if (b == 0) 
            return a
        return gcd(b, a % b)
    }

    def gcd(a: Array[Int]): Int{
        var ans = a[0]
        for(i in 1 until a.size)
            ans = gcd(ans, a[i])
        return ans
    }

    def lcm(a: Int, b: Int): Int{
        return a * (b / gcd(a, b))
    }

    def lcm(a: Array[Int]): Int{
        var ans = a[0]
    
        for (i in 1 until a.size) 
            ans = (((a[i] * ans)) / (gcd(a[i], ans))) 
    
        return ans
    }


    def getTotalX(a: Array[Int], b: Array[Int]): Int = {
        var l = lcm(a)
        var g = gcd(b)
        
        var count = 0
        var i = l
        var j = 2
        while(i <= g){
            if(g%i==0)
                count++
                
            i = l * j
            j++
        }
        return count
    }

}

object Solution {
    def main(args: Array[String]) {
        val printWriter = new PrintWriter(sys.env("OUTPUT_PATH"))

        val firstMultipleInput = StdIn.readLine.replaceAll("\\s+$", "").split(" ")

        val n = firstMultipleInput(0).toInt

        val m = firstMultipleInput(1).toInt

        val arr = StdIn.readLine.replaceAll("\\s+$", "").split(" ").map(_.trim.toInt)

        val brr = StdIn.readLine.replaceAll("\\s+$", "").split(" ").map(_.trim.toInt)

        val total = Result.getTotalX(arr, brr)

        printWriter.println(total)

        printWriter.close()
    }
}
